src/main/java/net/snowflake/client/core/SessionProperties.java

```diff
@@ -0,0 +1,442 @@
+package net.snowflake.client.core;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import net.snowflake.client.jdbc.ErrorCode;
+import net.snowflake.client.jdbc.SnowflakeType;
+
+public class SessionProperties {
```

Comment:
For this PR, could we have this be an abstract class instead and have the session inherit it.

Reply:
How about an interface instead? Then `SessionInterface` can extend it. This is more a 'has-a' relationship than a 'is-a' one.

Reply:
Ah sure, I didn't realize java let you do that now.

Reply:
So to be clear, the reason I want you to do that is so that we don't have to replace session.getX() with session.getProperties().getX() everywhere

Reply:
Yes, I figured. :) It's a reasonable suggestion.